### PR TITLE
chore: Support v1beta1 in CloudProvider interface

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

This PR migrates the CloudProvider interface to use the new `v1beta1` APIs. A change will have to be made in `aws/karpenter` to support this breaking change. This change is another one of the changes on the path to fully migrating the repository over to the new v1beta1 APIs.

**How was this change tested?**

`make presubmit`
`/karpenter snapshot`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: pkg/cloudprovider/cloudprovider.go [hunk 1]
```diff
@@ -61,7 +61,6 @@ import (
 
 func init() {
 	v1alpha5.NormalizedLabels = lo.Assign(v1alpha5.NormalizedLabels, map[string]string{"topology.ebs.csi.aws.com/zone": v1.LabelTopologyZone})
-	corev1beta1.NormalizedLabels = lo.Assign(corev1beta1.NormalizedLabels, map[string]string{"topology.ebs.csi.aws.com/zone": v1.LabelTopologyZone})
```
Comment (njtran) [inline pkg/cloudprovider/cloudprovider.go:64]: We don't need to add these into the normalized labels anymore?

File: pkg/cloudprovider/cloudprovider.go [hunk 2]
```diff
@@ -291,14 +287,14 @@ func (c *CloudProvider) resolveInstanceTypes(ctx context.Context, nodeClaim *cor
 	}
 	reqs := scheduling.NewNodeSelectorRequirements(nodeClaim.Spec.Requirements...)
 	return lo.Filter(instanceTypes, func(i *cloudprovider.InstanceType, _ int) bool {
-		return reqs.Compatible(i.Requirements) == nil &&
+		return reqs.Compatible(i.Requirements, lo.Ternary(nodeClaim.IsMachine, scheduling.AllowUndefinedWellKnownLabelsV1Alpha5, scheduling.AllowUndefinedWellKnownLabelsV1Beta1)) == nil &&
```
Comment (njtran) [inline pkg/cloudprovider/cloudprovider.go:290]: Can you remind me why we're adding more requirements checks here?

File: pkg/controllers/machine/garbagecollection/suite_test.go [hunk 3]
```diff
@@ -96,7 +96,7 @@ var _ = Describe("MachineGarbageCollection", func() {
 		nodeTemplate := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{})
 		provisioner := test.Provisioner(coretest.ProvisionerOptions{
 			ProviderRef: &v1alpha5.MachineTemplateRef{
-				APIVersion: v1alpha5.TestingGroup + "v1alpha1",
+				APIVersion: "testing.karpenter.sh/v1alpha1",
```
Comment (njtran) [inline pkg/controllers/machine/garbagecollection/suite_test.go:99]: Do we need to change this?
```suggestion
				APIVersion: "testing/v1alpha1",
```

File: pkg/controllers/machine/link/suite_test.go [hunk 4]
```diff
@@ -92,7 +92,7 @@ var _ = Describe("MachineLink", func() {
 		nodeTemplate = test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{})
 		provisioner = test.Provisioner(coretest.ProvisionerOptions{
 			ProviderRef: &v1alpha5.MachineTemplateRef{
-				APIVersion: v1alpha5.TestingGroup + "v1alpha1",
+				APIVersion: "testing.karpenter.sh/v1alpha1",
```
Comment (njtran) [inline pkg/controllers/machine/link/suite_test.go:95]: Same here

File: test/pkg/environment/common/environment.go [hunk 5]
```diff
@@ -92,7 +91,7 @@ func (env *Environment) Stop() {
 
 func NewConfig() *rest.Config {
 	config := controllerruntime.GetConfigOrDie()
-	config.UserAgent = fmt.Sprintf("%s-%s", v1alpha5.TestingGroup, project.Version)
+	config.UserAgent = fmt.Sprintf("testing.karpenter.sh-%s", project.Version)
```
Comment (njtran) [inline test/pkg/environment/common/environment.go:94]: Same here

File: test/suites/scale/deprovisioning_test.go [hunk 6]
```diff
@@ -43,7 +43,7 @@ import (
 )
 
 const (
-	deprovisioningTypeKey = v1alpha5.TestingGroup + "/deprovisioning-type"
+	deprovisioningTypeKey = "testing.karpenter.sh/deprovisioning-type"
```
Comment (njtran) [inline test/suites/scale/deprovisioning_test.go:46]: And here!

File: pkg/cloudprovider/cloudprovider.go (hunk 1 above)
Comment (jonathan-innis) [reply pkg/cloudprovider/cloudprovider.go:64]: Whoops, just a rebase miss

File: pkg/cloudprovider/cloudprovider.go (hunk 2 above)
Comment (jonathan-innis) [reply pkg/cloudprovider/cloudprovider.go:290]: It's not more compatibility checks. It's a change in the compatibility logic which assumes that keys that are undefined are denied by default, unless you specify an option that tells us to accept certain undefined keys

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: website/content/en/docs/concepts/node-templates.md [hunk 7]
```diff
@@ -579,7 +579,7 @@ Final merged UserData -
 <powershell>
 Write-Host "Running custom user data script"
 [string]$EKSBootstrapScriptFile = "$env:ProgramFiles\Amazon\EKS\Start-EKSBootstrap.ps1"
-& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022,testing.karpenter.sh/test-id=unspecified" --max-pods=110' -DNSClusterIP '10.0.100.10'
+& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022" --max-pods=110' -DNSClusterIP '10.0.100.10'
```
Comment (njtran) [inline website/content/en/docs/concepts/node-templates.md:582]: Don't we just need to rename? Why are we removing?
```suggestion
& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022,testing/test-id=unspecified" --max-pods=110' -DNSClusterIP '10.0.100.10'
```

File: website/content/en/preview/concepts/node-templates.md [hunk 8]
```diff
@@ -581,7 +581,7 @@ Final merged UserData -
 <powershell>
 Write-Host "Running custom user data script"
 [string]$EKSBootstrapScriptFile = "$env:ProgramFiles\Amazon\EKS\Start-EKSBootstrap.ps1"
-& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022,testing.karpenter.sh/test-id=unspecified" --max-pods=110' -DNSClusterIP '10.0.100.10'
+& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022" --max-pods=110' -DNSClusterIP '10.0.100.10'
```
Comment (njtran) [inline website/content/en/preview/concepts/node-templates.md:584]: same here
```suggestion
& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022,testing/test-id=unspecified" --max-pods=110' -DNSClusterIP '10.0.100.10'
```

File: website/content/en/v0.30/concepts/node-templates.md [hunk 9]
```diff
@@ -579,7 +579,7 @@ Final merged UserData -
 <powershell>
 Write-Host "Running custom user data script"
 [string]$EKSBootstrapScriptFile = "$env:ProgramFiles\Amazon\EKS\Start-EKSBootstrap.ps1"
-& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022,testing.karpenter.sh/test-id=unspecified" --max-pods=110' -DNSClusterIP '10.0.100.10'
+& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022" --max-pods=110' -DNSClusterIP '10.0.100.10'
```
Comment (njtran) [inline website/content/en/v0.30/concepts/node-templates.md:582]:
```suggestion
& $EKSBootstrapScriptFile -EKSClusterName 'test-cluster' -APIServerEndpoint 'https://test-cluster' -Base64ClusterCA 'ca-bundle' -KubeletExtraArgs '--node-labels="karpenter.sh/capacity-type=spot,karpenter.sh/provisioner-name=windows2022,testing/test-id=unspecified" --max-pods=110' -DNSClusterIP '10.0.100.10'
```

File: website/content/en/docs/concepts/node-templates.md (hunk 7 above)
Comment (jonathan-innis) [reply website/content/en/docs/concepts/node-templates.md:582]: This shouldn't have been in the docs to begin with. Seems odd that it's in there.

File: website/content/en/preview/concepts/node-templates.md (hunk 8 above)
Comment (jonathan-innis) [reply website/content/en/preview/concepts/node-templates.md:584]: Commented above

File: website/content/en/v0.30/concepts/node-templates.md (hunk 9 above)
Comment (jonathan-innis) [reply website/content/en/v0.30/concepts/node-templates.md:582]: Commented above

**Review by njtran — APPROVED** [review]

**Pull request merged**